34131771 - NULL help index section causes SIGSEGV in strcmp()

Filed upstream: https://github.com/OpenPrinting/cups/issues/378

--- a/cgi-bin/help-index.c	2022-05-03 15:20:50.591191518 +0200
+++ b/cgi-bin/help-index.c	2022-05-03 15:20:23.909699196 +0200
@@ -579,7 +579,7 @@
   */
 
   for (; node; node = (help_node_t *)cupsArrayNext(hi->nodes))
-    if (section && strcmp(node->section, section))
+    if (node->section && section && strcmp(node->section, section))
       continue;
     else if (filename && strcmp(node->filename, filename))
       continue;
